### ISSUE TITLE
Depending on specific configurations does not work

### ISSUE THREAD

**Issue by ansman** [dcendents/android-maven-gradle-plugin#39]
I'm writing a library with multiple modules that depend on each other.

My dependencies are set up like this:
```groovy
dependencies {
  debugCompile project(path: ":submodule", configuration: "debug")
  releaseCompile project(path: ":submodule", configuration: "release")
}
```

This results in the `<dependencies />` tag in the resulting `.pom` file to be empty.

I tried adding this to all modules' `build.gradle`:
```groovy
android {
    defaultPublishConfig "release"
    publishNonDefault true
}
```

This might be related to #11 and #32, I'm not sure but is there anything I can do about this?

**Comment by ansman** [comment]
I believe the problem comes from https://github.com/dcendents/android-maven-gradle-plugin/blob/master/src/main/java/org/gradle/api/plugins/AndroidMavenPlugin.java#L189

Only the dependencies from the `compile` configuration is considered. When I tried to add `releaseCompile` and `debugCompile` too it sort of works, though I get a duplicate dependency which is also lacking the configuration.

**Comment by ansman** [comment]
@dcendents Any thoughts on this?

**Comment by bpappin** [comment]
this might be related to #50.

**Comment by dcendents** [comment]
Is your project public so I get look at your configuration?

**Comment by ansman** [comment]
Not really, but it can be reproduced using a project with two modules with the main module having this:

```
dependencies {
  debugCompile project(path: ":submodule", configuration: "debug")
  releaseCompile project(path: ":submodule", configuration: "release")
}
```

and the submodule this:

```
android {
    defaultPublishConfig "release"
    publishNonDefault true
}
```

The problem is that `android-maven-gradle-plugin` simply doesn't look for `debugCompile` and `releaseCompile` dependencies.

**Comment by dcendents** [comment]
Please read my comment on issue #50 , I feel you are also trying to do something that is not natural in a maven repository.

Maven expects a single set of dependencies for all classifiers. If I include compile, debugCompile, releaseCompile and any other configuration you might define through flavors it will end up with a pom that contains everything, so it will also be wrong.

It's probably safer to keep using compile only, as it is the "default" configuration to use. Then if you have particular needs for a library, you might need to write a custom handler: https://docs.gradle.org/current/userguide/maven_plugin.html#sub:dependency_mapping

**Comment by ansman** [comment]
Well, releaseCompile might be nice for having different artifacts for debug and release (which is our use case).
It would result in two different artifact (one debug and one release) so the same artifact would never conflict.

**Comment by bpappin** [comment]
@ansman is is the plugin or the build that doesn't look for the dependencies?
in ticket #50 I reference a bug that apparently impacts this.
